google/cloud/status.cc

```diff
@@ -91,6 +71,103 @@ std::ostream& operator<<(std::ostream& os, StatusCode code) {
   return os << StatusCodeToString(code);
 }
 
+// Encapsulates the implementation of a non-OK status. OK Statuses are
+// represented by a nullptr Status::impl_, as an optimization for the common
+// case of OK Statuses. This class holds all the data associated with a non-OK
+// Status so we don't have to worry about bloating the common OK case.
+class Status::Impl {
+  using PayloadType = std::unordered_map<std::string, std::string>;
```

Comment:
Explicit access specifier?

Reply:
Actually, I wanted it to be private. It's only there to save some typing and readability in the body of `Impl`. I didn't intend to expose it to others as a usable member alias.